Tests/X509Tests/PolicyBuilderTests.swift

```diff
@@ -328,4 +408,97 @@ final class PolicyBuilderTests: XCTestCase {
             }
         }
     }
+
+    func testChainFailsPolicy_OneOf_empty() async {
+        await assertFailsToMeetPolicy {
+            OneOfPolicies {}
+        }
+    }
+
+    func testChainMeetsPolicy_OneOf_concatenation_both_valid() async {
+        await assertMeetsPolicy {
+            OneOfPolicies {
+                Policy(result: .meetsPolicy)
+                Policy(result: .meetsPolicy)
+            }
+        }
+    }
+
+    func testChainMeetsPolicy_OneOf_concatenation_first_valid() async {
+        await assertMeetsPolicy {
+            OneOfPolicies {
+                Policy(result: .meetsPolicy)
+                Policy(result: .failsToMeetPolicy(reason: ""))
+            }
+        }
+    }
+
+    func testChainMeetsPolicy_OneOf_concatenation_second_valid() async {
+        await assertMeetsPolicy {
+            OneOfPolicies {
+                Policy(result: .failsToMeetPolicy(reason: ""))
+                Policy(result: .meetsPolicy)
+            }
+        }
+    }
+
+    func testChainFailsToMeetPolicy_OneOf_concatenation_both_invalid() async {
+        await assertFailsToMeetPolicy {
+            OneOfPolicies {
+                Policy(result: .failsToMeetPolicy(reason: ""))
+                Policy(result: .failsToMeetPolicy(reason: ""))
+            }
+        }
+    }
+
+    func testChainMeetsPolicy_allOf() async {
+        await assertMeetsPolicy {
+            AllOfPolicies {
+                Policy(result: .meetsPolicy)
+                Policy(result: .meetsPolicy)
+            }
+        }
+        await assertFailsToMeetPolicy {
+            AllOfPolicies {
+                Policy(result: .meetsPolicy)
+                Policy(result: .failsToMeetPolicy(reason: ""))
+            }
+        }
+    }
+
+    func testChainMeetsPolicy_allOf_oneOf() async {
+        await assertMeetsPolicy {
+            OneOfPolicies {
+                AllOfPolicies {
+                    Policy(result: .meetsPolicy)
+                }
+            }
+        }
+        await assertFailsToMeetPolicy {
+            OneOfPolicies {
+                AllOfPolicies {
+                    Policy(result: .failsToMeetPolicy(reason: ""))
+                }
+            }
+        }
+        await assertMeetsPolicy {
+            OneOfPolicies {
+                Policy(result: .meetsPolicy)
+                Policy(result: .failsToMeetPolicy(reason: ""))
+                AllOfPolicies {
+                    Policy(result: .meetsPolicy)
+                    Policy(result: .meetsPolicy)
+                }
+            }
+        }
+        await assertFailsToMeetPolicy {
+            OneOfPolicies {
+                Policy(result: .failsToMeetPolicy(reason: ""))
+                AllOfPolicies {
+                    Policy(result: .meetsPolicy)
+                    Policy(result: .failsToMeetPolicy(reason: ""))
+                }
+            }
+        }
```

Comment:
Can we add tests for `if`, `if` + `else` and `if let` similar to the `PolicyBuilder` tests above?